flocker/cli/script.py

```diff
@@ -126,18 +119,55 @@ def __init__(self, ssh_configuration=None, ssh_port=22):
         self.ssh_configuration = ssh_configuration
         self.ssh_port = ssh_port
 
-    def _configure_ssh(self, deployment):
+    def main(self, reactor, options):
+        """
+        See :py:meth:`ICommandLineScript.main` for parameter documentation.
+
+        :return: A ``Deferred`` which fires when the deployment is complete or
+                 has encountered an error.
+        """
+        if options["nossh"]:
+            ready = succeed(None)
+        else:
+            ready = self._configure_ssh(
+                options["deployment_config"]["nodes"].keys())
+
+        body = dumps({"applications": options["application_config"],
+                      "deployment": options["deployment_config"]})
+        ready.addCallback(
+            lambda _: post(options["url"], data=body,
+                           headers={b"content-type": b"application/json"},
+                           persistent=False))
+
+        def fail(msg):
+            raise SystemExit(msg)
+
+        def got_response(response):
+            if response.code != OK:
+                d = json_content(response)
+                d.addCallback(
+                    lambda error: fail(error[u"description"] + u"\n"))
+                return d
+            else:
+                sys.stdout.write(_OK_MESSAGE)
+        ready.addCallback(got_response)
+        return ready
+
+    def _configure_ssh(self, hostnames):
         """
+        :param list hostnames: The addresses of the machines for which to
+            configure SSH keys.
+
         :return: A ``Deferred`` which fires when all nodes have been configured
-            with ssh keys.
+        with ssh keys.
```

Comment:
I wish I had a python mode for emacs that wrapped rst markup correctly in this case.  glyph's python-docstring-mode almost works but not quite...